### ISSUE TITLE
don't uppercase codeblocks inside `<h5>` headers

### ISSUE THREAD

**Issue by DetachHead** [squidfunk/mkdocs-material#7734]
### Description

`h5` headers convert the text to uppercase (#1522). i don't have an issue with this behavior, but i think by default it should leave any text inside a `<code>` block as-is, because in most cases code is case-sensitive.

### Related links

- previous issue #1522

that issue proposed removing the uppercasing entirely. my proposal is to just remove it inside `<code>` blocks

### Use Cases

API documentation generated by [mkdocstrings](https://mkdocstrings.github.io/) puts function/class names in headings

### Visuals

given the following markdown:
```md
##### Difference between `Never` and `NoReturn`
```

current:
![image](https://github.com/user-attachments/assets/5c693b5a-10a9-47f7-bd00-8809860f0ad4)

proposed:
![image](https://github.com/user-attachments/assets/93b2f990-feb0-4d9d-abab-aec8d8cfca81)

### Before submitting

- [X] I have read and followed the [change request guidelines](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/).
- [X] I have verified that [my idea is a change request and not a bug report](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/#its-not-a-bug-its-a-feature).
- [X] I have ensured that, to the best of my knowledge, [my idea will benefit the entire community](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/#benefit-for-the-community).
- [X] I have included relevant links to [the documentation](https://squidfunk.github.io/mkdocs-material/), related [issues](https://github.com/squidfunk/mkdocs-material/issues), and [discussions](https://github.com/squidfunk/mkdocs-material/discussions) to underline the need for my idea.

**Comment by DetachHead** [comment]
workaround: put this in your `extra.css`
```css
.md-typeset h5 code {
    text-transform: none;
}
```

**Comment by squidfunk** [comment]
Thanks for suggesting. I think this is a reasonable request, albeit it looks a little weird, but uppercase code is incorrect.

**Comment by squidfunk** [comment]
Fixed in b3f1eb27f with the patch suggested in https://github.com/squidfunk/mkdocs-material/issues/7734#issuecomment-2495380858.

**Comment by squidfunk** [comment]
Released as part of [9.5.46](https://squidfunk.github.io/mkdocs-material/changelog/#9.5.46).
